### ISSUE TITLE
Documentation structure

### ISSUE THREAD

**Issue by jasongrout** [jupyterlab/jupyterlab#7387]
I think this article has some really good points about documentation structure, and the differences between tutorials, how-tos, reference, and explanations: https://www.divio.com/blog/documentation/

Here is an example of it in practice: https://brachiograph.readthedocs.io/en/latest/#contents

**Comment by krassowski** [comment]
> including adding documentation on how to use Gitpod for JupyterLab development

There is already is a section on [Contributing from within the browser](https://jupyterlab.readthedocs.io/en/latest/developer/contributing.html#contributing-from-within-the-browser) which mentions Gitpod. It also mentions `jupyterlab-playground` which may enable hot-swapping some jupyterlab plugins from browser once Lab 4 lands with Lumino 2 (thanks to https://github.com/jupyterlab/lumino/pull/377) too.

I am not sure if we want to hide extensions development so deep as you are proposing (User guide → Reference guides  → JupyterLab extensions development); currently they are a top-level item:
   ![Screenshot from 2022-09-14 17-22-56](https://user-images.githubusercontent.com/5832902/190209507-c224fca1-614c-42e5-a77c-1cb19260c033.png)

I would advocate for keeping them separate as it crucial for JupyterLab to have robust extension community and make it easy for everyone to find this section ;)

On a general note, I would agree that the current https://jupyterlab.readthedocs.io/en/latest/developer/contributing.html has grown substantially and I see why you propose to split a summary out of it; I am not against, but personally I like having the ability to use ctrl + f to find items on a single pager rather than look though multiple pages. I guess there is merit in not overwhelming potential contributor too.

I would say that the "extras" like changelog and troubleshooting would be better placed at least one level higher (because when users are angry that  something broke or something changed in a new release they want to find these sections quick or will get angrier).

One thing that I love in numpy's (and [pandas](https://pandas.pydata.org/pandas-docs/stable/)) docs is this welcome screen:

![Screenshot from 2022-09-14 17-31-49](https://user-images.githubusercontent.com/5832902/190211873-45489c32-e179-4686-b900-5605edb46e78.png)

**Comment by trallard** [comment]
> There is already is a section on [Contributing from within the browser](https://jupyterlab.readthedocs.io/en/latest/developer/contributing.html#contributing-from-within-the-browser) which mentions Gitpod. It also mentions jupyterlab-playground which may enable hot-swapping some jupyterlab plugins from browser once Lab 4 lands with Lumino 2 (thanks to https://github.com/jupyterlab/lumino/pull/377) too.

yes, I did not count that as a complete sub-section since it is a bullet points list - and I am not sure folks can get up to speed with the content there. But could we turn that into a more extensive section on "alternative/lower friction" setups?

> I am not sure if we want to hide extensions development so deep as you are proposing (User guide → Reference guides → JupyterLab extensions development); currently, they are a top-level item:

This is something I was not sure of - it feels a bit o,dd in either place tbh. I agree that it is not 💯  Lab, but it is crucial for the ecosystem. As I said - this proposal might not be perfect and some of you have more context on why some things are the way they are.

> I would say that the "extras" like changelog and troubleshooting would be better placed at least one level higher (because when users are angry that something broke or something changed in a new release they want to find these sections quick or will get angrier).

You mean at the same level of User guides? It could definitely work - we could actually add a "Need help?" or similarly aimed section in the navbar


> One thing that I love in numpy's (and [pandas](https://pandas.pydata.org/pandas-docs/stable/)) docs is this welcome screen:

I do like this too and the [Gastby one](https://www.gatsbyjs.com/docs/) - we should definitely add something like that. I left this out of the original suggestion as it is a significant "new" item and would depend on the outcome of this issue to get right.

**Comment by afshin** [comment]
@krassowski:
> I like having the ability to use ctrl + f to find items on a single pager rather than look though multiple pages

One convention I've seen to address this while maintaining the structure is to add a link for a single-page version e.g., for downloading, `CTRL-F`, etc.

**Comment by krassowski** [comment]
> Another option is to keep the structure as is (Extensions dev into References) and keep the link in the navbar to ensure visibility/findability. WDYT?

I think it would work ok.

> > I would say that the "extras" like changelog and troubleshooting would be better placed at least one level higher (because when users are angry that something broke or something changed in a new release they want to find these sections quick or will get angrier).
>
> Do you mean at the same level as User guides? It could work - we could add a "Need help?" or similarly aimed section in the navbar.

Yes, that what I was suggesting. I would not personally search for changelog in "Need help?" but maybe it's a me thing. Looking at other projects, almost all I checked have "Release notes" section in the top level (pandas, numpy, matplotlib, scipy); sklearn does not (they have it under "more" which does not fully count).

**Comment by trallard** [comment]
Adding the changelog to the navbar helps. 

@afshin @jasongrout I'd really like to help with this reorg. Would it make sense to pop by a JupyterLab community meeting and share/discuss this to move forward?

**Comment by jasongrout** [comment]
> Would it make sense to pop by a JupyterLab community meeting and share/discuss this to move forward?

Yes, that would get a lot of eyes on it. Thanks!

**Comment by jtpio** [comment]
> Would it make sense to pop by a JupyterLab community meeting and share/discuss this to move forward?

Even a draft PR to get this started would be great, thanks!

**Comment by trallard** [comment]
Sounds good - I have had scheduling conflicts the past couple of weeks so have been unable to make it to the JLab calls
